allennlp/data/token_indexers/token_characters_indexer.py

```diff
@@ -34,23 +36,23 @@ def __init__(self,
         self._character_tokenizer = character_tokenizer
 
     @overrides
-    def count_vocab_items(self, token: str, counter: Dict[str, Dict[str, int]]):
-        for character in self._character_tokenizer.tokenize(token)[0]:
-            # If our character tokenizer is using byte encoding, the character might already be an
-            # int.  In that case, we'll bypass the vocabulary entirely.
-            if not isinstance(character, int):
-                counter[self._namespace][character] += 1
+    def count_vocab_items(self, token: Token, counter: Dict[str, Dict[str, int]]):
+        if token.text is None:
+            raise ConfigurationError('TokenCharactersIndexer needs a tokenizer that retains text')
+        for character in self._character_tokenizer.tokenize(token.text):
```

Comment:
I think you should leave some version of the comment that was here before, to explain the following lines

Reply:
Done.